allensdk/brain_observatory/behavior/data_objects/trials/trials.py

```diff
@@ -0,0 +1,192 @@
+from allensdk.brain_observatory.behavior.data_objects.metadata \
+    .behavior_metadata.equipment import \
+    Equipment
+from allensdk.brain_observatory.behavior.data_objects.rewards import Rewards
+from allensdk.brain_observatory.behavior.data_objects.trials.trial import Trial
+from allensdk.internal.brain_observatory.time_sync import OphysTimeAligner
+
+
+class Trials(DataObject, StimulusFileReadableInterface, NwbReadableInterface,
+             NwbWritableInterface):
+    def __init__(self, trials: pd.DataFrame):
+        super().__init__(name='trials', value=trials)
+
+    def to_nwb(self, nwbfile: NWBFile) -> NWBFile:
+        trials = self.value
+        order = list(trials.index)
+        for _, row in trials[['start_time', 'stop_time']].iterrows():
+            row_dict = row.to_dict()
+            nwbfile.add_trial(**row_dict)
+
+        for c in trials.columns:
+            if c in ['start_time', 'stop_time']:
+                continue
+            index, data = dict_to_indexed_array(trials[c].to_dict(), order)
+            if data.dtype == '<U1':  # data type is composed of unicode
+                # characters
+                data = trials[c].tolist()
+            if not len(data) == len(order):
+                if len(data) == 0:
+                    data = ['']
+                nwbfile.add_trial_column(
+                    name=c,
+                    description='NOT IMPLEMENTED: %s' % c,
+                    data=data,
+                    index=index)
+            else:
+                nwbfile.add_trial_column(
+                    name=c,
+                    description='NOT IMPLEMENTED: %s' % c,
+                    data=data)
+        return nwbfile
+
+    @classmethod
+    def from_nwb(cls, nwbfile: NWBFile) -> "Trials":
+        trials = nwbfile.trials.to_dataframe()
+        if 'lick_events' in trials.columns:
+            trials.drop('lick_events', inplace=True, axis=1)
+        trials.index = trials.index.rename('trials_id')
+        return Trials(trials=trials)
+
+    @classmethod
+    def from_stimulus_file(cls, stimulus_file: StimulusFile,
+                           stimulus_timestamps: StimulusTimestamps,
+                           licks: Licks,
+                           rewards: Rewards,
+                           monitor_delay: Optional[float] = None,
+                           sync_file: Optional[SyncFile] = None,
+                           equipment: Optional[Equipment] = None
+                           ) -> "Trials":
+        if monitor_delay is None:
+            if sync_file is None or equipment is None:
+                raise ValueError('Need sync file and equipment in order to '
+                                 'calculate monitor delay')
+            monitor_delay = cls._calculate_monitor_delay(sync_file=sync_file,
+                                                         equipment=equipment)
+            bsf = stimulus_file.data
```

Comment:
Should everything below here still be in the `if monitor_delay is None` block?

Reply:
Nope, thank you for catching it